File: Data Structures For Interview/JavaScript/Basic Programs/FactorialWithRecursion.js
function printFactorial(n){

    if(n==0){
        return 1;
    }
    else{
        return n*printFactorial(n-1);
    }
}

let n=4;
let factorial=printFactorial(n);
console.log("Factorial of "+n+" is :- "+factorial);

File: Data Structures For Interview/Java/Basic Programs/FactorialWithRecursion.java
public class FactorialWithRecursion {

    static int printFactorial(int n){

        if(n==0){
            return 1;
        }
        else{
            return (n*printFactorial(n-1));
        }
    }
    
    public static void main(String[] args) {
       
        int n=4;
        int factorial=printFactorial(4);
        System.out.println("Factorial of "+n+" is :- "+factorial);
    }
}
